x-pack/platform/plugins/shared/onechat/server/services/chat/utils/conversations.ts

```diff
@@ -65,10 +65,16 @@ export const updateConversation$ = ({
     roundCompletedEvent: roundCompletedEvents$,
   }).pipe(
     switchMap(({ title, roundCompletedEvent }) => {
+      // TODO: this should probably be handled by the conversation client
+      const { round, resumed = false } = roundCompletedEvent.data;
+      const updatedRound = resumed
+        ? [...conversation.rounds.slice(0, -1), round]
+        : [...conversation.rounds, round];
```

Comment:
This should probably move to the conversation client, but I'm waiting for a bigger refactor (the chat and agent services will need to get merged at some point and the runner needs some refactor too), so it's fine for now